### ISSUE TITLE
Upgrading a Ng1Component and using $element does not work

### ISSUE THREAD

**Issue by kevinverelst** [angular/angular#13912]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report 
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Trying to upgrade a ng1 component that uses inline templates and the $element service. The angular2 component will have NO children nodes.

**Expected behavior**
The child nodes should be available instantly (like in ng1)

**Minimal reproduction of the problem with instructions**
I'm running a ng1 app and use ngUpgrade to upgrade ng1 components to ng2 ones. The inline template is not instantly available.

*MINIMAL DEMO*
```
(function () {
    'use strict';

    angular.module('testComponent', []);

    angular.module('testComponent')
        .component('testComponent', {
            template: '<div><p>This is shitty</p></div>',
            controller: testComponentController
        })

    function testComponentController($element, $timeout) {
        console.log('Instant', $element.children());
        this.$onInit = function () {
            console.log('On Init', $element.children());
            $timeout(function () {
                console.log('After Timeout', $element.children());
            });
        }
    }
}
```

*OUTPUT NG1*
```
Instant [div]
On Init [div]
After Timeout [div]
```

*OUTPUT NG2*
```
Instant JQLite {}
On Init JQLite {}
After Timeout [div.ng-scope]
```

As you can see, the element is not available until the $timeout.

**What is the motivation / use case for changing the behavior?**
Gradually upgrade a ng1 application to ng2 instead of having to do a big bang

**Please tell us about your environment:**
Cordova android/ios/windows Phone hybrid angular 1 app trying to migrate to angular2
Using: 
- typescript 2.1.4
- Angular 2.4.1
- Angular 1.5.8
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
Typescript compiling to ES5

**Comment by dbpieter** [comment]
Can confirm...

Whereas a pure ng1 component has $element available in $onInit (as per the documentation). An upgraded ng1 component does not have $element available in $onInit until after sometime.

**Comment by gkalpak** [comment]
This sounds like an `upgrade` (aka `upgrade/dynamic`) issue, not an `upgrade/static` one.
@kevinverelst, @dbpieter: Can you confirm?

**Comment by kevinverelst** [comment]
I used 
```import { UpgradeModule } from '@angular/upgrade/static';```
as described in the [Upgrade documentation](https://angular.io/docs/ts/latest/guide/upgrade.html)

**Comment by gkalpak** [comment]
The title mentions `upgradeAdapter.upgradeNg1Component`, which is not an `upgrade/static` API 😕

**Comment by kevinverelst** [comment]
Fair enough, the title is a bit wrong (updated it) but the point is that the $element behaves differently in an ng1 component than in a ng2 component.

I'm using ```import {UpgradeComponent} from '@angular/upgrade/````to upgrade the component to a ng2 component

**Comment by gkalpak** [comment]
Sure. Just trying to figure out where the problem lies (it makes it easier to fix 😛)

**Comment by kevinverelst** [comment]
Any progress?

**Comment by gkalpak** [comment]
I am still looking into it. In the meantime, it doesn't seem right to try to access the content of the element in `$onInit()`. Using `$postLink` seems more appropriate for that purpose (and it [works as expected][1]).

[1]: http://plnkr.co/edit/xcTQVaOTCRQFJLaXj2Ig?p=preview

**Comment by gkalpak** [comment]
I have a potential fix for this in #14289 (that will make the compiling/instantiating/linking process more similar to AngularJS's in general).
